src/lib/ngsi/EntityId.cpp

```diff
@@ -140,7 +140,7 @@ std::string EntityId::toJson(void) const
 */
 std::string EntityId::check(RequestType requestType)
 {
-  if (id == "")
+  if (id == "" || ((id.find('\0')) != (std::string::npos)))
```

Comment:
The C++ check here is maybe too cryptic an potencially ineficient in a field that is going to be checked very ofter (a large set of operations in the API use entities id). Mabye @kzangeli have some C-based alternative that could be evaluated.

Reply:
Yes we can do it. I did it for an immediate fix. As we need it ASAP.
I will look for the workaround soon :P

Reply:
Yeah, this is kind of strange ... '\0' is simply 0, there is no difference.
And the check is to make sure there is no ZERO inside the string. There cannot be any zero inside a string in C. The zero is the marker that the string has ended ... Very cryptid indeedd. I understand nothing of this.  The functest tells me it works though ... Might be something weird about std::string but ... beats me.

What happens if you do this...?:

```
std::string s = "abcde";
char*   cP = (char*) s.c_str();

cP[2] = 0;
```

In C ... something similar would set `s` to "ab". Position 2 is now the end of the string ...
Looking forward to hearing an explanation of all this ... :-)

Reply:
Ah!  Could it have something to do with UTF?  "Funny" charsets ... ?